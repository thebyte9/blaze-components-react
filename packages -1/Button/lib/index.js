"use strict";
var __assign = (this && this.__assign) || function () {
    __assign = Object.assign || function(t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
            s = arguments[i];
            for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
                t[p] = s[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};
var __rest = (this && this.__rest) || function (s, e) {
    var t = {};
    for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p) && e.indexOf(p) < 0)
        t[p] = s[p];
    if (s != null && typeof Object.getOwnPropertySymbols === "function")
        for (var i = 0, p = Object.getOwnPropertySymbols(s); i < p.length; i++) {
            if (e.indexOf(p[i]) < 0 && Object.prototype.propertyIsEnumerable.call(s, p[i]))
                t[p[i]] = s[p[i]];
        }
    return t;
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
exports.__esModule = true;
var utils_1 = __importDefault(require("@blaze-react/utils"));
var react_1 = __importDefault(require("react"));
var Button = utils_1["default"](function (_a) {
    var _b;
    var disabled = _a.disabled, type = _a.type, children = _a.children, _c = _a.modifiers, modifiers = _c === void 0 ? [] : _c, classNames = _a.utils.classNames, attrs = __rest(_a, ["disabled", "type", "children", "modifiers", "utils"]);
    var formatedModifiers = modifiers
        .map(function (modifier) { return "button--" + modifier; })
        .join(" ");
    var buttonClassNames = classNames("button", (_b = {},
        _b[formatedModifiers] = !!modifiers,
        _b));
    return (react_1["default"].createElement("button", __assign({ disabled: disabled, className: buttonClassNames, type: type }, attrs), children));
});
var availableModifiers = {
    alert: "alert",
    back: "back",
    cta: "cta",
    dark: "dark",
    disabled: "disabled",
    fullWidth: "full-width",
    icon: "icon",
    light: "light",
    link: "link",
    outline: "outline",
    plain: "plain",
    rounded: "rounded",
    small: "small"
};
Button.availableModifiers = availableModifiers;
Button.defaultProps = {
    children: "",
    disabled: false,
    type: "button"
};
exports["default"] = Button;
