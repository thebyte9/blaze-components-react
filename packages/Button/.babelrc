{
  "presets": ["@babel/preset-env", "@babel/preset-react", "@babel/preset-typescript"],
  "plugins": [
    [
      "@babel/plugin-transform-runtime",
      {
        "regenerator": true
      }
    ],
    "@babel/plugin-syntax-jsx",
    "@babel/proposal-class-properties",
    "@babel/proposal-object-rest-spread"
  ]
}
