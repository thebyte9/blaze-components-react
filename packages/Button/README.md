
## Description

Custom button styles for actions in forms, dialogs, and more with support for multiple sizes, states, and more.

## Usage

* Simple

```js
<Button>Button</Button>
```

* With modifier

```js
<Button modifiers="plain">Plain</Button>

<Button modifiers="small">Small</Button>
```

* Combining modifiers

```js
<Button modifiers="outline rounded">
  Rounded/Outline
</Button>
```

* Disabled

```js
<Button disabled>Alert</Button>
```

* Type submit

```js
<Button submit>Send</Button>
```

## API

##### Buttons can receive a number of `props` as follow:


| NAME   | TYPE | DEFAULT | OPTIONS |
| :---  | :---:  | :---: | ------- |
| modifiers | String | empty | rounded, outline, alert, cta, light, dark, disabled, icon, small, full-width, back, link |
| submit | Boolean | false | 
| disabled | Boolean | false | 
| children | single/array of nodes | null | 
