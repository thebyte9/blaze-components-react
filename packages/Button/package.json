{
  "name": "@blaze-react/button",
  "version": "0.6.0",
  "description": "Custom button styles for actions in forms, dialogs, and more with support for multiple sizes, states, and more.",
  "main": "./src/index.ts",
  "scripts": {
    "build": "NODE_ENV=production webpack --config webpack.config.js",
    "build-storybook": "build-storybook -c .storybook -o docs-build",
    "lint": "eslint '*/**/*.{ts,tsx}' --ignore-path ../../.eslintignore --quiet --fix",
    "test": "jest",
    "test:coverage": "jest --coverage",
    "pre-publish": "yarn build && yarn lint && yarn test"
  },
  "keywords": [],
  "author": "byte9",
  "license": "MIT",
  "peerDependencies": {
    "prop-types": "^15.7.2",
    "react": "^17.0.2",
    "react-dom": "^17.0.2"
  },
  "devDependencies": {
    "webpack": "^5.28.0",
    "webpack-cli": "^4.6.0"
  }
}
