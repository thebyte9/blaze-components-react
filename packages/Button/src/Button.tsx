import React from 'react';
import { buildClassNames } from '@blaze-react/utils';

type TType = 'button' | 'submit' | 'reset';
export interface IButtonProps {
  disabled?: boolean;
  children?: unknown;
  utilities?: string;
  modifiers?: string[];
  type?: TType;
}

export const Button = ({ children, utilities = '', ...rest }: IButtonProps): JSX.Element => {
  const { modifiers = [], disabled, type = 'button' } = rest;

  const DeprecatedButton = () => {
    console.warn('Modifiers will be deprecated in the near future. You should use CSS utilities classes instead');

    const formatedModifiers: string = modifiers.map((modifier) => `button--${modifier}`).join(' ');
    const buttonClassNames: string = buildClassNames('button', {
      [formatedModifiers]: !!modifiers,
    });

    return (
      <button disabled={disabled} className={buttonClassNames} type={type} {...rest}>
        {children}
      </button>
    );
  };

  const BlazeButton = (
    <button type="button" className={utilities} {...rest}>
      {children}
    </button>
  );

  return utilities !== '' ? BlazeButton : DeprecatedButton();
};
