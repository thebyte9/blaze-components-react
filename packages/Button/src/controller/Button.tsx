import React from 'react';
import { DisplayIconProps, IButtonProps } from '../types';
import { ButtonView } from '../view/ButtonView';
import { Icon } from '@blaze-react/icon';
import { useComponentLogic } from '../hooks/useComponentLogic';

export const Button = ({
  label,
  type,
  variant,
  iconOnly = false,
  disabled = false,
  icon,
  displayIcon = DisplayIconProps.NoIcon,
  overrides = [],
  children,
  theme,
  onClick,
}: IButtonProps): JSX.Element => {
  const { display, classes } = useComponentLogic({
    componentVariant: variant,
    theme: theme,
    displayIcon,
    iconOnly: iconOnly,
    overrides: overrides,
  });

  return (
    <ButtonView classes={classes} type={type} disabled={disabled} onClick={onClick}>
      <Icon display={display} iconOnly={iconOnly} label={label} icon={icon} />
      {children}
    </ButtonView>
  );
};
