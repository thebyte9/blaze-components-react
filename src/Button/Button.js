import React from 'react';
import PropTypes from 'prop-types';

const Button = ({className, children}) => (
    <button onClick={() => console.log('click')} className={`${className}`} >
        {children}
    </button>
);

Button.propTypes = {
    children: PropTypes.oneOfType([
        PropTypes.arrayOf(PropTypes.node),
        PropTypes.node
    ]),
    className: PropTypes.string.isRequired,
};

export default Button;
