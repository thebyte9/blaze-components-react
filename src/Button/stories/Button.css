h1 {
  display: 'inline-block';
  border-bottom: '1px solid black';
}

table {
  text-align: 'center';
  border: '1px solid #dee2e6';
  width: '100%';
};

.showCaseDivStyles {
  display: 'flex';
  justify-content: 'space-around';
  flex-wrap: 'wrap';
  min-height: '10em';
};
