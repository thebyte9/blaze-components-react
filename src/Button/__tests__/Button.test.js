import React from 'react';
import expect from 'expect';
import { shallow } from 'enzyme';
import Button from '../Button';

const testProps = {
  className: 'button'
};

describe('Button component', () => {
  const wrapper = shallow(<Button {...testProps} />);

  test('should be defined', () => {
    expect(wrapper).toBeDefined();
  });

  test('should render without throwing an error', () => {
    expect(wrapper.is('.button')).toBe(true);
  });

  test('should be type button by default', () => {
    expect(wrapper.props().type).toBe('button');
  });

  test('should renders correctly (snapshot)', () => {
    expect(wrapper).toMatchSnapshot();
  });

  test('should be type submit when passing isSubmit prop', () => {
    expect(shallow(<Button isSubmit />).props().type).toBe('submit');
  });
});
