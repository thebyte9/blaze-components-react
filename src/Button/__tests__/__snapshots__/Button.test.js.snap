// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ForgotenPassword component renders correctly (snapshot) 1`] = `ShallowWrapper {}`;
